src/components/OrientationAwareIcon.tsx

```diff
@@ -17,32 +18,41 @@ export const OrientationAwareIcon = (props: Props) => {
   const { orientation, dimensions } = useContext(OrientationContext);
   const {
     Icon,
+    iconName,
     iconStyle,
     landscapeIconStyle,
     landscapeStyle,
     size = normalize(24),
     style
   } = props;
 
+  const SelectedIcon = Icon || IconComponent.NamedIcon;
+  const isNamedIcon = SelectedIcon === IconComponent.NamedIcon;
+
   const needLandscapeStyle =
     orientation === 'landscape' || dimensions.width > consts.DIMENSIONS.FULL_SCREEN_MAX_WIDTH;
 
-  // we need adjustments only on iOS, so otherwise just return the item with the usual props
-  // TODO: is this still needed with Expo 48?
-  if (!needLandscapeStyle || !(device.platform === 'ios')) {
-    return <Icon {...props} />;
-  }
+  const getIconProps = () => {
+    const baseProps = {
+      ...props,
+      ...(isNamedIcon && { name: iconName })
```

Comment:
if `isNamedIcon` is false, does `...` work or break?

Reply:
It works without a problem.

Reply:
Hmm. I haven't tested it this way. Then I will update the code this way.

```
const baseProps = {
  ...props,
  ...(isNamedIcon ? { name: iconName } : {})
};
```

Reply:
all right, yes, i think this is more secure, thanks